dss_client/test/test_dss_c_interface.c

```diff
@@ -18,13 +19,55 @@ int hexdump(unsigned char* buff, int size)
 	return 0;
 }
 
+int putRecursive(DSSClient c, char *basePath)
+{
+    int i;
+    char path[1000];
+    struct dirent *dp;
+    DIR *dir = opendir(basePath);
+
+    if (!dir){
+		return 0;
+	}
+
+    while ((dp = readdir(dir)) != NULL)
+    {
+        if (strcmp(dp->d_name, ".") != 0 && strcmp(dp->d_name, "..") != 0)
+        {
+            strcpy(path, basePath);
+            strcat(path, "/");
+            strcat(path, dp->d_name);
+            // printf("%s\n", path);
+			FILE* fp = fopen(path, "r");
+			if (fp == NULL) { 
+				return 0; 
+			} 
+			fseek(fp, 0L, SEEK_END); 
+			// calculating the size of the file 
+			long int fsize = ftell(fp); 
+			if (dp->d_type == DT_REG && fsize > 0){ // PutObject does not support empty files
+				if (PutObject(c, (void*) path, strlen(path), path) < 0 ){
+					printf("PutObject - %s failed.\n", path);
+					return -1;
+				}
+			}
+            if (putRecursive(c, path) < 0) return -1;
```

Comment:
rather than going into the function putRecursive to check whether dir or not, you can add else case. Since you are already checking regular file in the top and you have dp->d_type handy.